src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,85 @@
-export const Pagination = () => {};
+import React, { useMemo } from 'react';
+import classnames from 'classnames';
+
+type Props = {
+  total: number[];
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalResult = total.slice(
+    perPage * currentPage - perPage,
+    perPage * currentPage,
+  );
+
+  const existingPages = total.slice(0, Math.ceil(42 / perPage));
+
+  useMemo(() => onPageChange(total[0]), [perPage]);
```

Comment:
Use useMemo to memorize some value, not for function calls. You can use useEffect() for it